src/python/pants/engine/rules.py

```diff
@@ -28,9 +28,13 @@ class EngineAware(ABC):
     are expected to be overridden by concrete types implementing EngineAware.
     """
 
-    @abstractmethod
     def level(self) -> Optional[LogLevel]:
         """Overrides the level of the workunit associated with this type."""
+        return None
+
+    def info_message(self) -> Optional[str]:
```

Comment:
Because we're in the context of logging, "info" is overloaded here I think... maybe "result_message"? Or just "message"?